### PR TITLE
Add distribution reference

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**
I add a reference of distribution.
#4773

### CODE REVIEW

File: chainer/distribution.py [hunk 1]
```diff
@@ -8,11 +8,14 @@ class Distribution(object):
...
+    
+    1.Sampling random points.
+    
     2. Evaluating a probability-related function at a given realization value.
-        (e.g., probability density function, probability mass function)
+    (e.g., probability density function, probability mass function)
+    
     3. Obtaining properties of distributions.
-        (e.g., mean, variance)
+    (e.g., mean, variance)
```
Comment (toslunar) [inline chainer/distribution.py]: Insert a space after `1.` and fix indents of second lines of `2.` and `3.`:
```
    1. Sampling random points.
    
    2. Evaluating a probability-related function at a given realization value.
       (e.g., probability density function, probability mass function)
    
    3. Obtaining properties of distributions.
       (e.g., mean, variance)
```

Comment (toslunar) [reply chainer/distribution.py]: And remove spaces from empty lines:
```
    1. Sampling random points.

    2. Evaluating a probability-related function at a given realization value.
       (e.g., probability density function, probability mass function)

    3. Obtaining properties of distributions.
       (e.g., mean, variance)
```

File: docs/source/reference/distributions.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+   :toctree: generated/
+   :nosignatures:
+   
+   chainer.distributions.Normal
+
+
+Functionals of distribution
+---------------------------
+
+.. module:: chainer
```
Comment (beam2d) [inline docs/source/reference/distributions.rst]: Use `currentmodule` instead of `module`.

File: docs/source/reference/index.rst [hunk 3]
```diff
@@ -22,3 +22,4 @@ Reference
    graph
    caffe
    check
+   distributions
```
Comment (beam2d) [inline docs/source/reference/index.rst]: I feel it's better to place this document to some upper place (e.g. bellow `links`). (maybe we need more structure to this reference section, but that is another issue)

**Comment by YoshikawaMasashi** [comment]
OK, I changed.

**Comment by YoshikawaMasashi** [comment]
OK. I opened a PR #4899.

**Pull request merged**